backstop_data/engine_scripts/puppet/price-set/preview-price-field-modal.js

```diff
@@ -1,14 +1,13 @@
 'use strict';
 
+const Page = require('../page-objects/crm-page.js');
+
 module.exports = async (engine, scenario, viewport) => {
-  const priceFieldsEmpty = await engine.$('#crm-main-content-wrapper .messages');
+  const page = new Page(engine, scenario, viewport);
 
   await require('./view-and-edit-price-fields')(engine, scenario, viewport);
-  if(priceFieldsEmpty) {
-    console.warn('No Price list present!')
-    console.log('Taking the "Price Fields List" Page screenshot..');
-  } else {
+  await page.checkIfPriceFieldsAreEmpty('Fields').then(async () => {
     await engine.click('a[title="Preview Price"]');
     await engine.waitForSelector('.CRM_Price_Form_Preview');
-  }
+  }, async () => {});
```

Comment:
Why do we need that?

Reply:
Yes! Otherwise the error thrown from the `checkIfPriceFieldsAreEmpty` fn will not be handled (caught) and will be passed to the backstop directly and which will eventually kill the scenario and throw backstop error.

Reply:
@igorpavlov : I do a bit of googling and stackoverflowing. And found out that `catch()` is *not deprecated* and in fact like you said it is a better way of writing and handling error cases.
<img width="676" alt="screen shot 2018-06-27 at 12 14 07 pm" src="https://user-images.githubusercontent.com/3340537/41959388-6bf16204-7a0a-11e8-8af2-04c2df1e2d03.png">

Also `catch()` is the part of ES6 specifications.

References - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises 
<img width="785" alt="screen shot 2018-06-27 at 1 06 54 pm" src="https://user-images.githubusercontent.com/3340537/41959589-074a1638-7a0b-11e8-9ce9-b6c585190905.png">

Updated the PR too. Please check.
Sorry! My bad!